src/Vector3.h

```diff
@@ -121,11 +126,7 @@ class Vector3
 	{
 		// Perform a bitwise comparison of the contents - we want to know whether this object is exactly equal
 		// To perform EPS-based comparison, use the EqualsEps() function
-		return (
-			(memcmp(&x, &a_Rhs.x, sizeof(x)) == 0) &&
-			(memcmp(&y, &a_Rhs.y, sizeof(y)) == 0) &&
-			(memcmp(&z, &a_Rhs.z, sizeof(z)) == 0)
-		);
+		return !((x != a_Rhs.x) || (y != a_Rhs.y) || (z != a_Rhs.z));
```

Comment:
By using memcmp you are assuming that the object you are comparing is POD. Since Vector3 is a template, you can't guarantee that. 
Then there's things like positive zero and negative zero for floats which should compare equal, but have a different bit representation.